csrc/ir/nodes.cpp

```diff
@@ -4859,9 +4859,16 @@ std::vector<PolymorphicValue> SdpaBwdOp::evaluate(
     const std::vector<PolymorphicValue>& inputs) const {
   // Backward tensor inputs: grad_input, query, key, value, output, logsumexp,
   // max_q/k
+  // Temporary handling of DID parallelization see
+  // https://github.com/NVIDIA/Fuser/issues/2563
+  bool handle_device_dim = inputs.at(1).as<at::Tensor>().dim() == 5;
   std::vector<at::Tensor> bwd_inputs;
   for (auto idx : c10::irange(6)) {
-    bwd_inputs.emplace_back(inputs.at(idx).as<at::Tensor>());
+    auto in_tensor = inputs.at(idx).as<at::Tensor>();
+    if (handle_device_dim) {
+      in_tensor = in_tensor.squeeze(0);
+    }
+    bwd_inputs.emplace_back(in_tensor);
```

Comment:
I know this didn't come from you, but would you mind changing it to `push_back`? https://abseil.io/tips/112